apps/server/test/lexical/server/code_intelligence/entity_test.exs

```diff
@@ -480,30 +478,240 @@ defmodule Lexical.Server.CodeIntelligence.EntityTest do
     end
   end
 
+  describe "struct resolve/2" do
+    test "succeeds when the cursor is on the %", %{project: project} do
+      code = ~q[
+        |%MyStruct{}
+      ]
+
+      assert {:ok, {:struct, MyStruct}, resolved_range} = resolve(project, code)
+      assert resolved_range =~ ~S[%«MyStruct»{}]
+    end
+
+    test "succeeds when the cursor is in an alias", %{project: project} do
+      code = ~q[
+        %My|Struct{}
+      ]
+
+      assert {:ok, {:struct, MyStruct}, resolved_range} = resolve(project, code)
+      assert resolved_range =~ ~S[%«MyStruct»{}]
+    end
+
+    test "succeeds when the cursor is on the opening bracket", %{project: project} do
+      code = ~q[
+        %MyStruct|{}
+      ]
+
+      assert {:ok, {:struct, MyStruct}, resolved_range} = resolve(project, code)
+      assert resolved_range =~ ~S[%«MyStruct»{}]
+    end
+
+    test "succeeds when the struct spans multiple lines", %{project: project} do
+      code = ~q[
+        %MyStruct.|Nested{
+          foo: 1,
+          bar: 2
+        }
+      ]
+
+      assert {:ok, {:struct, MyStruct.Nested}, resolved_range} = resolve(project, code)
+      assert resolved_range =~ ~S[%«MyStruct.Nested»{]
+    end
+
+    test "excludes trailing module segments", %{project: project} do
+      code = ~q[
+        %My|Struct.Nested{}
+      ]
+
+      assert {:ok, {:struct, MyStruct}, resolved_range} = resolve(project, code)
+      assert resolved_range =~ ~S[%«MyStruct».Nested{}]
+    end
+
+    test "expands current module", %{project: project} do
+      code = ~q[
+            defmodule Example do
+              %|__MODULE__{}
+            end
+          ]
+
+      assert {:ok, {:struct, Example}, resolved_range} = resolve(project, code)
+      assert resolved_range =~ ~S[  %«__MODULE__»{}]
+    end
```

Comment:
another odd case would be `__MODULE__.Substruct|{}`, which has occurred in the past.

how about cases with aliased modules 
```elixir
`q[
  alias Something.Else 
  %Else.Another|{}
 ]
```
should resolve `Something.Else.Another`